src/dos/program_more.cpp

```diff
@@ -60,15 +61,79 @@ bool MORE::ParseCommandLine(MoreOutputFiles &output)
 	cmd->FillVector(params);
 
 	// Check if specified tabulation size
-	if (!params.empty()) {
-		const auto &param = params[0];
-		if ((starts_with("/t", param) || starts_with("/T", param)) &&
-		    (param.length() == 3) && (param.back() >= '1') &&
-		    (param.back() <= '9')) {
-			// FreeDOS extension - custom TAB size
-			output.SetTabSize(static_cast<uint8_t>(param.back() - '0'));
+	while (!params.empty()) {
+		auto is_option_extended = [](const std::string& param) {
+			return param == "/e" || param == "/E";
+		};
+		auto is_option_clear = [](const std::string& param) {
+			return param == "/c" || param == "/C";
+		};
+		auto is_option_expand_form_feed = [](const std::string& param) {
+			return param == "/p" || param == "/P";
+		};
+		auto is_option_squish = [](const std::string& param) {
+			return param == "/s" || param == "/S";
+		};
+		auto is_option_start_line = [](const std::string& param) {
+			if ((param.length() < 2) || (param[0] != '+')) {
+				return false;
+			}
+			for (size_t i = 1; i < param.length(); ++i) {
+				if (!std::isdigit(param[i])) {
+					return false;
+				}
+			}
+			return true;
+		};
+		auto is_option_tab_size = [](const std::string& param) {
+			return (param.length() == 3) &&
+			       (starts_with("/t", param) || starts_with("/T", param)) &&
+			       (param.back() >= '1') && (param.back() <= '9');
+		};
+
+		const auto& param = params[0];
+		if (is_option_extended(param)) {
+			output.SetOptionExtendedMode(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_clear(param)) {
+			output.SetOptionClear(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_expand_form_feed(param)) {
+			output.SetOptionExpandFormFeed(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_squish(param)) {
+			output.SetOptionSquish(true);
+			params.erase(params.begin());
+			continue;
+		}
+		if (is_option_start_line(param)) {
+			int32_t line_num = 0;
```

Comment:
There's a lot of hand-done command-line parsing here; I guess the `CommandLine` functionality was insufficient:

``` c++
bool FindExist(char const * const name,bool remove=false);
bool FindHex(char const * const name,unsigned int & value,bool remove=false);
bool FindInt(char const * const name,int & value,bool remove=false);
bool FindString(char const * const name,std::string & value,bool remove=false);
bool FindCommand(unsigned int which,std::string & value);
bool FindStringBegin(char const * const begin,std::string & value, bool remove=false);
bool FindStringRemain(char const * const name,std::string & value);
bool FindStringRemainBegin(char const * const name,std::string & value);
bool GetStringRemain(std::string & value);
int GetParameterFromList(const char* const params[], std::vector<std::string> & output);
void FillVector(std::vector<std::string> & vector);
unsigned int GetCount(void);
void Shift(unsigned int amount=1);
```

Would it make sense to augment what you need into the CommandLine parser? I'm hoping that `MORE.COM`'s CLI parsing needs aren't far more advanced that the parent program itself, which uses CommandLine. 

Ideally we can normalize/standardize all the CLI handling at some point behind some library or `CommandLine`(the goal be less about code refactoring and more about guaranteeing consistent CLI handling - once everything is using the same parsing backend).

Just my uninformed thought process.. if it's all really needed - then so be it!

Reply:
Initially I intended to require switches before file names (wouldn't be possible with these functions) - but it seems Windows is more tolerant, so we should be too. I'll rework the command line parsing.

But `/Tn` and `+nnn` (where `n` and `nnn` are integers, without spaces before them) will probably need additional command-line parsing functions - I'll investigate it, but not today.

Reply:
Command line parsing reworked.